doc/sphinx/src/metadata.rst

```diff
@@ -0,0 +1,282 @@
+
+Flag collections can be constructed from a C++
+standard library container of ``MetadataFlag`` objects, or simply a
+comma separated list of them. For example:
+
+.. code:: cpp
+
+   using parthenon::Metadata;
+   using parthenon::MetadataFlag;
+   using FS_t = Metadata::FlagCollection
+   // Constructor from a container
+   FS_t set1(std::vector<MetadataFlag>{Metadata::Cell, Metadata::Face});
+   // Constructor from a comma separated list
+   FS_t set2(Metadata::Requires, Metadata::Overridable);
+
+By default constructor arguments go into the ``Intersections`` property
+of the ``FlagCollection``. However, if a container is passed into the
+constructor, you can also pass in an optional boolean flag to specify
+whether or not you want to match **any** flags instead of **all**
+flag. This adds the constructor arguments to the ``Unions``
+property of the ``FlagCollection``.
+
+.. code:: cpp
+
+   // Implicit construction form a container, which
+   // requests EITHER the following flags instead of BOTH
+   FS_t set2({Metadata::Independent, Metadata::FillGhost}, true);
+
+The flags contained in the ``Unions``, ``Intersections``, and
+``Exclusions`` properties of the ``FlagCollection`` can be extracted via
+equivalently named accessors, which return a ``std::set``. For
+example:
+
+.. code:: cpp
+
+   const std::set<MetadataFlag> &u = set1.Unions();
+   const std::set<MetadataFlag> &i = set1.Intersections();
+   const std::set<MetadataFlag> &e = set1.Exclusions();
+
+For the most part, you should not need these accessors. They are used
+by Parthenon internal functions, such as variable and meshblock
+packing, to compute the correct variables to pack.
+
+You can add flags to these property fields with the ``TakeUnion``,
+``TakeIntersection``, and ``Exclude`` methods. These methods take
+either a standard library container of metadata flags, or another
+``FlagCollection`` instance.
+
+The ``FlagCollection`` class supports algebraic operations, although they are
+not entirely consistent with standard arithmetic order of
+operations. In particular:
+
+.. code:: cpp
+
+   // this could also be auto s = set1 || set2;
+   auto s = set1 + set2;
+
+produces a set s with the a unions field which is the set union of the
+union fields of set1 and set2. However,
+
+.. code:: cpp
+
+   // this could also be s = set1 && set2;
+   auto s = set1 * set2;
+
+Produces a set s with a "unions" field of set1 and an intersections
+field containing the original intersections of set1, and the
+intersections of set2.
+
+.. code:: cpp
+
+   auto s = Set1 - Set2
+
+Produces a set s with the "unions" and "intersections" fields of set1
+and an exclusion field containing set1's exlcusion field as well as
+ALL THREE fields (union, intersection, exclusion) contained by set2.
+
+This feels unintuitive, but it makes expressions like
+
+.. code:: cpp
+
+   auto s = {Flag1, FLag2} * {Flag3, Flag4} - {Flag5, Flag6}
```

Comment:
FLag -> Flag

Reply:
Sorry... maybe I am still confused on the logic here, this only returns 

```
EITHER Flag1 or Flag2 AND Flag3 AND Flag4 and NOT Flag5 or Flag6
```
if `{Flag1, Flag2}` was constructed via

```FS_t my_set({Flag1, Flag2}, true);```

and `{Flag3, Flag4}` was constructed via

```FS_t my_set({Flag1, Flag2}, false);``` or ```FS_t my_set({Flag1, Flag2});```

Is that right?

Reply:
Ah yeah I was sloppy here. Yes `set1` needs to be constructed with `FS_t my_set1({Flag1, Flag2}, true)`. `set2` needs to be constructed with the `use_union` flag disabled, as you noted. But that's the default, so it could be constructed as `FS_t({Flag1, Flag2}, false)`, `FS_t({Flag1, Flag2})`, or `FS_t(Flag1, Flag2)`.

There's also an object-oriented interface, which I forgot to document, so the same could be achieved with:
```C++
FS_t flags;
flags.TakenUnion(std::vector<MetadataFlag>{Flag1, Flag2});
flags.TakeIntersection(std::vector<MetadataFlag>{Flag3, Flag4});
flags.Exclude(std::vector<MetadataFlag>{Flag5, Flag6});
```
I can also make it so you don't need to pass in a container to make that work. I'll update accordingly...

Reply:
Done.